pkg/proxy/util/conntrack.go

```diff
@@ -28,10 +29,23 @@ import (
 
 const noConnectionToDelete = "0 flow entries have been deleted"
 
-// DeleteServiceConnections uses the conntrack tool to delete the conntrack entries
+func isIPv6(ip string) bool {
+	netIP := net.ParseIP(ip)
+	return netIP != nil && netIP.To4() == nil
+}
+
+func parametersWithFamily(isIPv6 bool, parameters ...string) []string {
+	if isIPv6 {
+		parameters = append(parameters, "-f", "ipv6")
+	}
+	return parameters
+}
+
+// ClearUDPConntrackForIP uses the conntrack tool to delete the conntrack entries
 // for the UDP connections specified by the given service IP
 func ClearUDPConntrackForIP(execer exec.Interface, ip string) error {
-	err := ExecConntrackTool(execer, "-D", "--orig-dst", ip, "-p", "udp")
+	parameters := parametersWithFamily(isIPv6(ip), "-D", "--orig-dst", ip, "-p", "udp")
```

Comment:
why not just do something like:

```
func familyFlag(ip string) string {
    if isIPv6(ip) {
        return "--family=ipv6"
    }
    return "--family=ipv4"
}
```

and then call that in all cases?  Seems cleaner.

Reply:
I'll approve anyway, but if thsi is cleaner, I'd appreciate it.